dexec/cmd.go

```diff
@@ -158,6 +165,19 @@ func (c *Cmd) Start() error {
 		c.osCancel()
 	default:
 	}
+
+	// On Windows when you send "os.Interrupt" (really syscall.CTRL_BREAK_EVENT) it goes to
+	// *all* processes in the process group.  So if we're going to be sending it ourselves, put
+	// the child process in its own group.  However, we'll also need to propegate it if
+	// something else sends it to our whole group, for consistency with the vanilla behavior.
+	var windowsInterruptCh chan os.Signal
+	if runtime.GOOS == "windows" && c.ctx != dcontext.HardContext(c.ctx) {
+		windowsInterruptCh := make(chan os.Signal, 10)
+		signal.Notify(windowsInterruptCh, os.Interrupt)
+		procAttr := sysProcAttrForNewGroup
+		c.SysProcAttr = &procAttr
```

Comment:
I quess a check is needed here to verify that `SysProcAttr` isn't already set by the caller and then just add the flag if it is?

Reply:
Ugg, I definitely under-thought about this.